src/Client.php

```diff
@@ -243,6 +243,14 @@ protected function getExceptionMap()
     }
 
     /**
+     * @return LoggerInterface
+     */
+    public function getLogger()
```

Comment:
Based on the discussion downstream. Do you think this has merits on its own or only for testing? If it's the latter I suggest we refactor `getClient()` a tiny bit.